tests: libc: Update character output testcases to solve coverity issues

yingmingx opened the pull request:
1. Refine some testcases for testing some character output function.
remove the code to reference NULL.
2. Delete some negative testcases which will not invoke fatal error.

Fixes #33822
Fixes #33803
Fixes #33801
Fixes #33800
Fixes #33790

Signed-off-by: Ying ming <mingx.ying@intel.com>

enjiamai reviewed (changes requested):
I guess we better not delete too many test cases until we can provide better test cases instead of it. And the NULL parameter test case, I think we can keep them. Some violations might be false positive. 

My suggestion is if you are targeting to fix the Coverity issue, just to correct/delete the violating test cases.

yingmingx commented on tests/lib/sprintf/src/main.c:
```diff
@@ -729,42 +728,23 @@ void test_sprintf_string(void)
 
 	ret = fprintf(stdout,  "%d", i);
 	zassert_equal(ret, 1, "fprintf failed!");
 
```
In this function, there are two part of code segment. The first one is testing fprintf with file pointer stdout , the second one is test printf. So left the testcase name unchanged because fprintf with stdout is equal to printf.

enjiamai replied:
Ok, got it.

yingmingx commented on tests/lib/sprintf/src/main.c:
```diff
@@ -729,42 +728,23 @@ void test_sprintf_string(void)
-	ztest_set_fault_valid(true);
-	ret = fprintf(NULL,  "%d", i);
-	zassert_not_equal(ret, 1, "fprintf failed!");
 }
```
This testcase was removed because it didn't invoke fatal error.

enjiamai replied:
I guess that means you can just remove 765, and make this false positive.

yingmingx replied:
Got it!

Commit:
tests: libc: Update character output testcases to solve coverity issues

1. Refine some testcases for testing some character output function.
2. Delete some negative testcases which will not invoke fatal error.

Fixes #33822
Fixes #33803
Fixes #33801
Fixes #33800
Fixes #33790

Signed-off-by: Ying ming <mingx.ying@intel.com>

yingmingx commented on tests/lib/sprintf/src/main.c:
```diff
@@ -792,25 +772,8 @@ void test_vfprintf(void)
-	zassert_equal(ret, 15, "vfprintf \"3\" failed");
-
-	ret = WriteFrmtd_vf(stdout,  "", 3);
-	zassert_equal(ret, 0, "vfprintf \"3\" failed");
```
I'll keep the testcase and change the parameter in WriteFrmtd_vf to fix coveity issues.

yingmingx commented on tests/lib/sprintf/src/main.c:
```diff
@@ -838,17 +801,8 @@ void test_vprintf(void)
 	zassert_equal(ret, 5, "vprintf \'a\' failed");
-
-	ret = WriteFrmtd_v("11", 'a');
-	zassert_equal(ret, 2, "vprintf \'a\' failed");
```
I keep the testcase. Please take another look, thanks!

enjiamai reviewed (approved):
Thanks for the update.  No other problem from my side.

KangJianX reviewed (approved):
Fixed coverity issues. LGTM!
